tests/unit/db/feed_tests.py

```diff
@@ -0,0 +1,544 @@
+            raw_content.append(raw_line)
+        changes = json.loads(''.join([unicode_(x) for x in raw_content]))
+        if self.cloudant_test:
+            self.assertSetEqual(
+                set(changes.keys()), set(['results', 'last_seq', 'pending']))
+        else:
+            self.assertSetEqual(
+                set(changes.keys()), set(['results', 'last_seq']))
+        results = list()
+        for result in changes['results']:
+            self.assertSetEqual(set(result.keys()), set(['seq', 'changes', 'id']))
+            results.append(result)
+        expected = set(['julia000', 'julia001', 'julia002'])
+        self.assertSetEqual(set([x['id'] for x in results]), expected)
+        self.assertTrue(str(changes['last_seq']).startswith('3'))
+        self.assertIsNone(feed.last_seq)
+
+    def test_get_normal_feed_default(self):
+        """
+        Test getting content back for a "normal" feed without feed option
+        """
+        self.populate_db_with_documents(3)
+        feed = Feed(self.session, self.changes_url)
+        changes = list()
+        for change in feed:
+            self.assertSetEqual(set(change.keys()), set(['seq', 'changes', 'id']))
+            changes.append(change)
+        expected = set(['julia000', 'julia001', 'julia002'])
+        self.assertSetEqual(set([x['id'] for x in changes]), expected)
+        self.assertTrue(str(feed.last_seq).startswith('3'))
+
+    def test_get_normal_feed_explicit(self):
+        """
+        Test getting content back for a "normal" feed using feed option
+        """
+        self.populate_db_with_documents(3)
+        feed = Feed(self.session, self.changes_url, feed='normal')
+        changes = list()
+        for change in feed:
+            self.assertSetEqual(set(change.keys()), set(['seq', 'changes', 'id']))
+            changes.append(change)
+        expected = set(['julia000', 'julia001', 'julia002'])
+        self.assertSetEqual(set([x['id'] for x in changes]), expected)
+        self.assertTrue(str(feed.last_seq).startswith('3'))
+
+    def test_get_continuous_feed(self):
+        """
+        Test getting content back for a "continuous" feed
+        """
+        self.populate_db_with_documents()
+        feed = Feed(self.session, self.changes_url, feed='continuous')
+        changes = list()
+        for change in feed:
+            self.assertSetEqual(set(change.keys()), set(['seq', 'changes', 'id']))
+            changes.append(change)
+            if len(changes) == 100:
+                feed.stop()
+        expected = set(['julia{0:03d}'.format(i) for i in range(100)])
+        self.assertSetEqual(set([x['id'] for x in changes]), expected)
+        self.assertIsNone(feed.last_seq)
+        # Compare continuous with normal
+        normal = Feed(self.session, self.changes_url)
+        self.assertSetEqual(
+            set([x['id'] for x in changes]), set([n['id'] for n in normal]))
+
+    def test_get_longpoll_feed(self):
```

Comment:
No because the heartbeat is just a way for us to drop out of the continuous feed and complete the test.  The actual heartbeat test is a few tests down.